version: '2'

services:
  jupyter:
    image: jupyter/tensorflow-notebook:45f07a14b422
    command: >
     jupyter notebook
     --NotebookApp.token=''
     --NotebookApp.password=''
    # removing token & password to enable traefik auth
    environment:
      - 'JUPYTER_ENABLE_LAB=yes'
    volumes:
      - './jupyter/jupyter/config:/root/.jupyter/'
      - './jupyter/notebooks:/home/jovyan/'
    networks:
      - 'srv'
    restart: always
    labels:
      - 'traefik.enable=true'
      - 'traefik.frontend.rule=Host:jupyter.${SITE}'
      - 'traefik.port=8888'
      - 'traefik.frontend.auth.basic.users=${USERS}'
      # - 'traefik.frontend.auth.forward.address=http://authelia:9091/api/verify?rd=https://authelia.${SITE}/'
      # - 'traefik.frontend.auth.forward.trustForwardHeader=true'
      # - 'traefik.frontend.auth.forward.authResponseHeaders=Remote-User,Remote-Groups,Remote-Name,Remote-Email'
